Readme & local dev dependencies

flannoo opened the pull request:
Hi,

I've been looking into contributing to this project to add some additional APIs to query on-chain data.
Figured this might be a good start for contributing:

- Added Readme to get started with local development
- Added "serverless-offline" plugin & dependencies to be able to run the project locally without an AWS account
- Added docker file to run a standalone opensearch instance to test locally

Let me know your thoughts on this :-)

tbekas reviewed (changes requested):
I like the `README.md` and the usage of the `serverless-offline` plugin. When it comes to the docker-compose however, we prefer using [skaffold](https://skaffold.dev/) for local development as it allows us to combine multiple parts of the system like snapshot-streaming and L0 and L1 cluster in a single environment that may be distributed across many physical nodes (in a kubernetes cluster). Opensearch is already deployed by [this skaffold configuration](https://github.com/Constellation-Labs/snapshot-streaming/blob/develop/skaffold.yaml) in snapshot-streaming and it's endpoint is binded to `localhost:4510`. Therefore I would rather like to see a config that connects to Opensearch deployed by existing skaffold configuration rather than an Opensearch instance deployed by docker-compose running in isolation from the rest of the system.

flannoo commented:
> I like the `README.md` and the usage of the `serverless-offline` plugin. When it comes to the docker-compose however, we prefer using [skaffold](https://skaffold.dev/) for local development as it allows us to combine multiple parts of the system like snapshot-streaming and L0 and L1 cluster in a single environment that may be distributed across many physical nodes (in a kubernetes cluster). Opensearch is already deployed by [this skaffold configuration](https://github.com/Constellation-Labs/snapshot-streaming/blob/develop/skaffold.yaml) in snapshot-streaming and it's endpoint is binded to `localhost:4510`. Therefore I would rather like to see a config that connects to Opensearch deployed by existing skaffold configuration rather than an Opensearch instance deployed by docker-compose running in isolation from the rest of the system.

Makes sense! I changed it accordingly:
- Removed docker compose file
- Updates to make it work with the snapshot-streaming skaffold environment (changed opensearch instance port in env.yml to 4510 & changed serverless offline port to 3001, since default is using port 3000 and that was giving conflicts with the grafana instance in the skaffold environment)

tbekas reviewed (approved):
LGTM
